src/components/data/data-utils.js

```diff
@@ -234,7 +242,9 @@ export const EntityUploader = ({ onSuccess, onDismiss, namespace, name, entityTy
     setUploading(true)
     try {
       const workspace = Ajax().Workspaces.workspace(namespace, name)
-      await (useFireCloudDataModel ? workspace.importEntitiesFile : workspace.importFlexibleEntitiesFile)(file)
+      const { jobId } = await (useFireCloudDataModel ? workspace.importEntitiesFile : workspace.importFlexibleEntitiesFileAsync)(file)
```

Comment:
I believe that right here, `file.size` will give you the size, in bytes, of the file-to-be-uploaded.

For a follow-on PR, or in this one if you feel like tackling it, I _thiiiink_ we can do something like:
```
      let filesize = Number.MAX_SAFE_INTEGER
      try {
        filesize = file.size
      } catch (err) {
        // noop
      }
      if (filesize < 524288) { // 512k, let's choose a good number, I chose this arbitrarily
        await (useFireCloudDataModel ? workspace.importEntitiesFile : workspace.importFlexibleEntitiesFile)(false, file)
      } else {
        const { jobId } = await (useFireCloudDataModel ? workspace.importEntitiesFile : workspace.importFlexibleEntitiesFile)(true, file)
        asyncImportJobStore.update(Utils.append({ targetWorkspace: { namespace, name }, jobId }))
        notifyDataImportProgress(jobId)
      }
```